.github/workflows/rust.yml

```diff
@@ -1,4 +1,4 @@
-name: Rust
+name: All
```

Comment:
Nit: can we rename this file to something like `workspace.yml` or `common.yml` or something? And a comment somewhere explaining "these tests operate on the entire workspace. they mostly check that everything builds and looks good"